### DISCUSSION TITLE
Ability to have inactive volunteers

### DISCUSSION THREAD

**Discussion by simply-emily** [CareTogether/CareTogetherCMS#137] [Ideas]
We need to be able to deactivate volunteers temporarily with a note in the system if they aren't currently able/wanting to serve

**Comment by LarsKemmann** [comment]
Thanks for the suggestion!

We had a somewhat similar idea submitted before (https://github.com/CareTogether/CareTogetherCMS/discussions/97), so let's try to clarify the differences between the two.

Would this toggle apply to a single volunteer in a family, or an entire family at once? Or does that depend on the role? (I'm which case that's what the previous suggestion was basically saying.)

Would it be more of a priority to have an overall active/inactive toggle, or a role-by-role toggle?

**Comment by simply-emily** [comment]
I think it would likely be best to be able to inactivate each individual adult volunteer-- perhaps by role as well. Could a volunteer potentially be able to be made inactive as a host family, but one spouse in the family still serve as a family coach, for example? 

So I guess the conclusion of that would be that it would be great for it to be a role-by-role toggle. :-) I'd like to hear what others think, though-- I may be missing something.

**Reply by LarsKemmann** [reply]
Based on this I think I'm going to combine this with #186 and #97 - it's really all the same underlying concept, just with maybe a different "reason": inactive (not currently serving in a role), opt out (not currently interested in serving in a role), denied (not currently allowed to serve in a role).

The way this works, when the volunteer is reactivated/re-opted-in/reinstated (respectively), CareTogether will automatically resume any approval requirements calculations, including putting the host family on hold if an adult child wants to start serving for example.

**Reply by LarsKemmann** [reply]
@AGNMM18 See my comment above - this will merge three different requests that have come up over the past couple of weeks.

Development on this feature request is being tracked in #180.
